pkg/controller/daemon/daemoncontroller.go

```diff
@@ -683,6 +740,48 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *api.Node, ds *exte
 	return fit
 }
 
+func (dsc *DaemonSetsController) needsRollingUpdate(ds *extensions.DaemonSet, daemonPods []*api.Pod, numUnavailable, numNodes int) (*api.Pod, bool) {
+	maxUnavailable, err := intstr.GetValueFromIntOrPercent(&ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable, numNodes, true)
+	if err != nil {
+		glog.Errorf("Invalid value for MaxUnavailable: %v", err)
+		return nil, false
+	}
+	if numUnavailable >= maxUnavailable {
+		glog.V(4).Infof("Number of unavailable DaemonSet pods: %d, is equal to or exceeds allowed maximum: %d", numUnavailable, maxUnavailable)
+		return nil, false
+	}
+
+	needsUpdate := true
+	var podToRetain *api.Pod
+	newPodTemplateSpecHash := podutil.GetPodTemplateSpecHash(ds.Spec.Template)
```

Comment:
Eventually, I think we should use a different hashing algorithm (we are going to change it for Deployments too). @smarterclayton suggested fnv